libs/SalesforceReact/SalesforceReact/Classes/SFNetReactBridge.m

```diff
@@ -50,13 +54,30 @@ @implementation SFNetReactBridge
     NSString* path = [argsDict nonNullObjectForKey:kPathArg];
     NSDictionary* queryParams = [argsDict nonNullObjectForKey:kQueryParams];
     NSDictionary* headerParams = [argsDict nonNullObjectForKey:kHeaderParams];
+    NSDictionary* fileParams = [argsDict nonNullObjectForKey:kfileParams];
     
     SFRestRequest* request = [SFRestRequest requestWithMethod:method path:path queryParams:queryParams];
+    
+    // Custom headers
     [request setCustomHeaders:headerParams];
     if (endPoint) {
         [request setEndpoint:endPoint];
     }
     
+    // Files post
+    if (fileParams) {
+        // File params expected to be of the form:
+        // {<fileParamNameInPost>: {fileMimeType:<someMimeType>, fileUrl:<fileUrl>, fileName:<fileNameForPost>}}
+        for (NSString* fileParamName in fileParams) {
+            NSDictionary* fileParam = fileParams[fileParamName];
+            NSString* fileMimeType = [fileParam nonNullObjectForKey:kFileMimeType];
+            NSString* fileUrl = [fileParam nonNullObjectForKey:kFileUrl];
+            NSString* fileName = [fileParam nonNullObjectForKey:kFileName];
+            NSData* fileData = [NSData dataWithContentsOfURL:[NSURL URLWithString:fileUrl]];
+            [request addPostFileData:fileData paramName:fileParamName fileName:fileName mimeType:fileMimeType];
```

Comment:
If the send request fails, we send back the error to javascript (see line 83 below)